Shared/DefaultsKeys.swift

```diff
@@ -34,20 +34,51 @@ extension DefaultsKeys {
     static func addFavoriteFood(_ name: String) {
         if !isFavoriteFood(name) {
             Defaults[\.favoriteFoods].append(name)
+            #if os(iOS)
+            DispatchQueue.global(qos: .background).async {
+                setLocations(name: name)
+            }
+            #endif
         }
     }
 
     static func removeFavoriteFood(_ name: String) {
         Defaults[\.favoriteFoods].removeAll(where: { item in
             item == name
         })
+        #if os(iOS)
+        Defaults[\.favoriteFoodLocations].removeValue(forKey: name)
+        #endif
     }
 
     static func isFavoriteFood(_ name: String) -> Bool {
         Defaults[\.favoriteFoods].contains(name)
     }
 
     #if os(iOS)
+    var favoriteFoodLocations: DefaultsKey<[String: [String]]> { .init("favoriteFoodLocations", defaultValue: [:]) }
+
+    static func updateFoodLocations(completion: @escaping () -> Void) {
```

Comment:
I think we usually say `_ completion`, though it shouldn't matter since this is a trailing closure.